polygeist/test/polygeist-opt/parallellicm.mlir

```diff
@@ -95,43 +174,42 @@ func.func @scf_parallel_nohoist1(%arg0: memref<?xf32>, %arg1: index, %arg2: inde
 }
 
 // -----
+
+// COM: Test LICM on affine.for loops.
 module {
-func.func private @use(f32) 
-// COM: Ensure loop invariant load is hoisted.
-func.func @affine_parallel_hoist1(%arg0: memref<?xf32>, %arg1: index, %arg2: index, %arg3 : index, %arg4 : index, %arg5: index, %arg6: index) {
-  // CHECK:       func.func @affine_parallel_hoist1(%arg0: memref<?xf32>, %arg1: index, %arg2: index, %arg3: index, %arg4: index, %arg5: index, %arg6: index) {
-  // CHECK-NEXT:     %cst = arith.constant 0.000000e+00 : f32
-  // CHECK-NEXT:     %alloca = memref.alloca() : memref<f32>
-  // CHECK-NEXT:     memref.store %cst, %alloca[] : memref<f32>
-  // CHECK-NEXT:     affine.if #set(%arg1, %arg2, %arg5, %arg3, %arg4, %arg6) {
-  // CHECK-NEXT:       %0 = memref.load %alloca[] : memref<f32>
-  // CHECK-NEXT:       affine.parallel (%arg7, %arg8) = (max(%arg1, %arg2), %arg5) to (min(%arg3, %arg4), %arg6) {
-  // CHECK-NEXT:         func.call @use(%0) : (f32) -> ()
-  // CHECK-NEXT:       }
-  // CHECK-NEXT:     }
+// CHECK: #set = affine_set<()[s0, s1] : (s1 - s0 - 1 >= 0)>
```

Comment:
Why is it not `(s0 < s2)`?

Reply:
Because the expression  ub - lb > 0 is equivalent to ub - lb -1 >= 0 given the values are integers.